src/server/pkg/hashtree/interface.go

```diff
@@ -46,34 +46,58 @@ const (
 )
 
 // HashTree is the signature of a hash tree provided by this library. To get a
-// new hash tree, see hashtree.NewHashTree().
+// new HashTree, create an OpenHashTree with NewHashTree(), modify it, and then
+// call Finish() on it.
 type HashTree interface {
-	// PutFile appends data to a file (and creates the file if it doesn't exist)
-	PutFile(path string, blockRefs []*pfs.BlockRef) error
-
-	// PutDir creates a directory (or does nothing if one exists)
-	PutDir(path string) error
-
-	// DeleteFile deletes a regular file or directory (along with its children).
-	DeleteFile(path string) error
+	// Open makes a deep copy of the HashTree and returns the copy
+	Open() OpenHashTree
 
-	// Get retrieves the contents of a regular file
+	// Get retrieves a file.
 	Get(path string) (*NodeProto, error)
 
 	// List retrieves the list of files and subdirectories of the directory at
 	// 'path'.
 	List(path string) ([]*NodeProto, error)
 
-	// Glob returns a list of files and directories that match 'pattern'
+	// Glob returns a list of files and directories that match 'pattern'.
 	Glob(pattern string) ([]*NodeProto, error)
 
+	// Serialize serializes a HashTree so that it can be persisted. Also see
+	// Deserialize(bytes).
+	Serialize() ([]byte, error)
```

Comment:
Deserialize is exported, but it's not part of the interface (otherwise you'd have to create a finished HashTree somehow, and then call `h.Deserialize(bytes)` to replace this contents, instead of `h := Deserialize(bytes)`